File: ApemanPlugin/src/main/kotlin/apeman_core/features_extraction/calculators/candidate/NumSwitchOperatorsCandidatesCalculator.java
package apeman_core.features_extraction.calculators.candidate;

import apeman_core.base_entities.FeatureType;
import apeman_core.pipes.CandidateWithFeatures;
import com.intellij.psi.JavaRecursiveElementVisitor;
import com.intellij.psi.PsiSwitchStatement;

import java.util.ArrayList;

public class NumSwitchOperatorsCandidatesCalculator extends AbstractNumCandidateCalculator {

    public NumSwitchOperatorsCandidatesCalculator(ArrayList<CandidateWithFeatures> candidates) {
        super(candidates, FeatureType.NUM_SWITCH);
    }

    @Override
    public JavaRecursiveElementVisitor createVisitor() {
        return new NumSwitchOperatorsCandidatesCalculator.Visitor();
    }

    private class Visitor extends CandidateVisitor {

        @Override
        public void visitSwitchStatement(PsiSwitchStatement statement) {
            super.visitSwitchStatement(statement);

            if (!isInsideMethod)
                return;
            updateCounters();
        }
    }
}

File: ApemanPlugin/src/main/kotlin/apeman_core/features_extraction/calculators/candidate/NumSwitchOperatorsCandidatesCalculator.kt
package apeman_core.features_extraction.calculators.candidate

import apeman_core.base_entities.ExtractionCandidate
import apeman_core.base_entities.FeatureType
import com.intellij.psi.PsiSwitchStatement

class NumSwitchOperatorsCandidatesCalculator(candidates: List<ExtractionCandidate>
) : AbstractNumCandidateCalculator(candidates, FeatureType.NUM_SWITCH) {

    override fun createVisitor(methodCandidates: List<ExtractionCandidate>) = Visitor()

    inner class Visitor : AbstractNumCandidateCalculator.CandidateVisitor() {

        override fun visitSwitchStatement(statement: PsiSwitchStatement) {
            super.visitSwitchStatement(statement)

            if (!isInsideMethod)
                return
            updateCounters()
        }
    }
}
